spec/models/thredded/post_spec.rb

```diff
@@ -85,105 +81,142 @@ module Thredded
-      @post.filter = 'markdown'
+      @post.content = <<-MARKDOWN.strip_heredoc
+        # Header
 
-      expect(@post.filtered_content(view_context))
-          .to eq("<h1>Header</h1>\n\n<p><a href=\"http://www.google.com\">http://www.google.com</a></p>")
+        http://www.google.com
+      MARKDOWN
+      expected_html = <<-HTML.strip_heredoc
+        <h1>Header</h1>
+        <p><a href="http://www.google.com">http://www.google.com</a></p>
+      HTML
+
+      resulting_parsed_html = parsed_html(@post.filtered_content(view_context))
+      expected_parsed_html  = parsed_html(expected_html)
+
+      expect(resulting_parsed_html).to eq(expected_parsed_html)
     end
 
     it 'performs some syntax highlighting in markdown' do
-      input = "this is code
+      @post.content = <<-MARKDOWN.strip_heredoc
+        this is code
 
-      def hello; puts 'world'; end
+            def hello; puts 'world'; end
 
-  right here"
+        right here
+      MARKDOWN
+      expected_html = <<-HTML.strip_heredoc.strip
+        <p>this is code</p>
 
-      expected_output = "<p>this is code</p>\n\n<pre><code>  def hello; puts 'world'; end\n</code></pre>\n\n<p>right here</p>"
+        <pre><code>def hello; puts 'world'; end
+        </code></pre>
 
-      @post.content = input
-      @post.filter = 'markdown'
+        <p>right here</p>
+      HTML
+      resulting_parsed_html = parsed_html(@post.filtered_content(view_context))
+      expected_parsed_html  = parsed_html(expected_html)
 
-      expect(@post.filtered_content(view_context)).to eq expected_output
+      expect(resulting_parsed_html).to eq(expected_parsed_html)
     end
 
     it 'links @names of members' do
       Thredded.user_path = ->(user) { "/whois/#{user}" }
+      post_content = 'for @sam but not @al or @kek. And @joe.'
       sam = build_stubbed(:user, name: 'sam')
       joe = build_stubbed(:user, name: 'joe')
-      post = build_stubbed(:post,
-                           content: 'for @sam but not @al or @kek. And @joe.')
-      expect(post).to receive(:readers_from_user_names).with(%w(sam al kek joe)).and_return([sam, joe])
-      expectation = '<p>for <a href="/whois/sam">@sam</a> but not @al or @kek. And <a href="/whois/joe">@joe</a>.</p>'
+      post = build_stubbed(:post, content: post_content)
+      expected_html = '<p>for <a href="/whois/sam">@sam</a> but not @al or @kek. And <a href="/whois/joe">@joe</a>.</p>'
+
+      expect(post).to receive(:readers_from_user_names)
+        .with(%w(sam al kek joe))
+        .and_return([sam, joe])
 
-      expect(post.filtered_content(view_context)).to eq expectation
+      expect(post.filtered_content(view_context)).to eq expected_html
     end
 
     def parsed_html(html)
-      Nokogiri::HTML::DocumentFragment.parse(html).to_hash
+      Nokogiri::HTML::DocumentFragment.parse(html) { |config| config.noblanks }
+        .to_html
+        .gsub(/^\s*/, '')
+        .gsub(/\s*$/, '')
+        .gsub(/^$\n/, '')
```

Comment:
Agreed. I wish there was. I think the use case is narrow and approachable enough, though, where someone patient enough could write a regex that does the job. Unfortunately I was/am tired and didn't want to jump through those hoops more than just chaining these gsubs, haha.